network/types.go

```diff
@@ -38,6 +38,12 @@ type Network struct {
 	// container's interfaces if a pair is created, specifically in the case of type veth
 	// Note: This does not apply to loopback interfaces.
 	TxQueueLen int `json:"txqueuelen,omitempty"`
+
+	// HairpinNAT specifies if hairpin NAT should be enabled on the virtual interface
+	// bridge port in the case of type veth
+	// Note: This is unsupported on some systems.
+	// Note: This does not apply to loopback interfaces.
+	HairpinNAT bool `json:"hairpin_nat"`
```

Comment:
Actually that thing not called `HairpinNAT`. It is rather `HairpinMode` or just `Hairpin`